apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/Priority.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright 2020 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Google.Cloud.Spanner.Data
+{
+    /// <summary>
+    /// The relative priority for requests.
+    /// </summary>
+    public enum Priority
+    {
+        /// <summary>
+        /// Executes the request with unspecified priority.
+        /// </summary>
+        Unspecified,
```

Comment:
Should this value be surfaced here? I think I would prefer making the properties on transactions and commands nullable, and convert the null to the `UNSPECIFIED` in the proto. If you look at the proto generated code, `UNSPECIFIED` is being ignored (i.e. treated as a `null`) for all `RequestOptions` operations (`WriteTo`s, `Merge`, `CalculateSize`, even `GetHashCode` ...).
An alternative to making the properties null, and given that it says in the proto docs that `UNSPECIFIED` means `HIGH`, is to default `Priority` to `High` always. I still prefer making the priority properties nullable.
@jskeet What do you think?

Reply:
I can see pros and cons to this. There's something appealing about keeping it mirroring the proto enum, but equally Amanda's point is fairly convincing.

I don't much mind either way - I'd just point out that if this is removed, `Low` needs to be given the explicit value of 1 to keep parity with the proto enum.

Reply:
My preference would be to keep this as closely aligned with the proto definition as possible, and keep the `Unspecified` value accessible for users. That way, anyone reading the Spanner gRPC documentation will see the same values here as in that part of the documentation.
Making it nullable would also be an option, although it would not be my first choice for the reasons above.

I don't think we should use `High` as the default. The current documentation states that `Unspecified` will default to `High`. However, if the backend team decides to change that in the future, it would require a manual intervention in the client as well.